compaction/compaction_manager.cc

```diff
@@ -1470,8 +1470,8 @@ class compaction_manager::cleanup_sstables_compaction_task : public compaction_m
 bool needs_cleanup(const sstables::shared_sstable& sst,
                    const dht::token_range_vector& sorted_owned_ranges,
                    schema_ptr s) {
-    auto first = sst->get_first_partition_key();
-    auto last = sst->get_last_partition_key();
+    const auto& first = sst->get_first_partition_key();
+    const auto& last = sst->get_last_partition_key();
     auto first_token = dht::get_token(*s, first);
     auto last_token = dht::get_token(*s, last);
```

Comment:
You can further optimize this by using `sst->get_first_decorated_key()` and use not re-decorate the key here.

Reply:
That's possible if the partitioner didn't change (which is most likely).
But I'll add a check for that, just in case.